src/sparselm/model/_base.py

```diff
@@ -152,78 +342,109 @@ def _gen_objective(self, X: ArrayLike, y: ArrayLike):
                 Covariate/Feature matrix
             y (ArrayLike):
                 Target vector
+            beta (cp.Variable):
+                cp.Variable representing the estimated coefs_
+            parameters (SimpleNamespace): optional
+                SimpleNamespace with cp.Parameter objects
+            auxiliaries (SimpleNamespace): optional
+                SimpleNamespace with auxiliary cvxpy objects
 
         Returns:
-            cvpx Expression
+            cvxpy Expression
         """
-        return None
+        return
 
-    def _gen_constraints(self, X: ArrayLike, y: ArrayLike):
+    def _generate_constraints(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        beta: cp.Variable,
+        parameters: Optional[SimpleNamespace] = None,
+        auxiliaries: Optional[SimpleNamespace] = None,
+    ) -> Optional[list[cp.constraints]]:
         """Generate constraints for optimization problem.
 
         Args:
             X (ArrayLike):
                 Covariate/Feature matrix
             y (ArrayLike):
                 Target vector
+            beta (cp.Variable):
+                cp.Variable representing the estimated coefs_
+            parameters (SimpleNamespace): optional
+                SimpleNamespace with cp.Parameter objects
+            auxiliaries (SimpleNamespace): optional
+                SimpleNamespace with auxiliary cvxpy objects
 
         Returns:
-            list of cvpx constraints
+            list of cvxpy constraints
         """
         return None
 
-    def _initialize_problem(self, X: ArrayLike, y: ArrayLike):
-        """Initialize cvxpy problem from the generated objective function.
+    def generate_problem(self, X: ArrayLike, y: ArrayLike) -> None:
```

Comment:
Why do we expose "generate_problem" as a public method to CVXRegressor? Shouldn't we only expose the "fit" method?

Reply:
I decided to expose it to enable advanced usage for example for a user to generate all cvxpy objects and then possibly add additional constraints. In order to do so, users must have access to the `canonicals_` generated here. A great example would be if we want to implement ground-state preservation constraints in CEs

Reply:
Can't we implement additional constraints with an exposed method to modify the self.canonicals_? Exposing generate_problems still does not seem to be very standard to me. For ground-state preservation, I think we can just add in tools a function that returns ground-state preservation constraints, and use the abovementioned public modification method to modify the canonicals.

Reply:
Also please consider whether to implement ground-state preservation as a part of CVXEstimator methods or just as a tool to be attached.

Reply:
Yes, by "implementing" I mean allowing the functionality to do so. I do not want any source code implementations of anything that is CE-specific. sparse-lm is a linear regression package and should be kept solely that.

Reply:
That being said, users (ie us) can more easily include additional constraints (such as ground state preservation in our work) by having access to the `canonicals_` which is generated in `generate_problem`. 

I'm not sure what you mean by a method to modify `canonicals_`. These still need to be generated before being able to modify them, and for the example of additional constraints, a user may need to access certain objects in `canonicals` (ie such as `beta` or `z0` in MIQP constraints)

Reply:
for this we would still need to implement something like a `add_constraints` method. However, that would be in addition to the `generate_problem` method, since as  I mentioned most cases will need access to  `canonicals`.